sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/modules/JGitAdapter.java

```diff
@@ -0,0 +1,66 @@
+package com.mercedesbenz.sechub.wrapper.prepare.modules;
+
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JGitAdapter {
+
+    public void clonePublicRepository(GitContext contextGit) {
+        clone(contextGit.getLocation(), contextGit.getUploadDirectory(), contextGit.isCloneWithoutHistory(), null, null);
+    }
+
+    public void clonePrivateRepository(GitContext contextGit, String username, String password) {
+        clone(contextGit.getLocation(), contextGit.getUploadDirectory(), contextGit.isCloneWithoutHistory(), username, password);
+    }
```

Comment:
As discussed, you could just have one method here. The private method clone could then look for credentials inside the GitContexts credentials Map and decide if credentials need to be set of not.
```suggestion
    public void cloneRepository(GitContext contextGit) {
        clone(contextGit);
    }
```